### ISSUE TITLE
newline

### ISSUE THREAD

**Issue by rcseacord** [mbmcmullen27/effective-c#1]
Nice job on this!  I forked this and noticed that at least one C source file (presumably more) fails to end in a newline.  

The C standard specifies that a C file should end with a newline (C23 CWD, 5.1.1.2 Translation phases, paragraph 2.) and that a last line without a newline yields undefined behavior:

Each instance of a backslash character (\) immediately followed by a new-line character is
deleted, splicing physical source lines to form logical source lines. Only the last backslash on
any physical source line shall be eligible for being part of such a splice. **A source file that is
not empty shall end in a new-line character**, which shall not be immediately preceded by a
backslash character before any such splicing takes place.

**Comment by mbmcmullen27** [comment]
Thanks! I fixed this the quick way in pr #2 with a little bash. I mostly made this repo to as a way to keep notes while I read the book, but if you have any other thoughts or ideas I'd be happy to hear them.
```
for i in ./*/*.c; do echo -ne "\n" >> $i; done
for i in ./*/*.h; do echo -ne "\n" >> $i; done
```

**Comment by rcseacord** [comment]
Right now I'm trying to figure out how to make it all build using Visual Code on Ubuntu.

**Comment by mbmcmullen27** [comment]
I'm currently away and don't have a copy of the book with me, but I believe chapter 10 was the compiling and linking chapter and has separate build scripts in it, but all the other chapters can be built with Make now that I merged #4. Some fail, and I marked those as "*.fails" so they won't block the build action you added.  If you have the repo cloned you can see them with:
```
echo */*.fails
```
I can investigate those further when I'm back and have a copy of the book on hand.
I also amended the build workflow to add the compiled exercises as a build artifact.
